src/util/pointer.h

```diff
@@ -205,11 +211,13 @@ class SharedBuffer
 	{
 		m_size = buffer.getSize();
```

Comment:
(Idea) If you want, you could probably shave off some LoC by calling the lower level constructor:
```C++
SharedBuffer(const Buffer<T> &buffer) : SharedBuffer(*buffer, buffer.getSize())
{
	// no-op
}
```